src/components/_myHead.tsx

```diff
@@ -0,0 +1,40 @@
+import Head from "next/head";
+import { NextPage } from "next";
+
+interface MyHeadProps {
+  title?: string;
+  description?: string;
+}
+
+const MyHead: NextPage<MyHeadProps> = ({ title, description }) => {
+  const siteName = "Kintai";
+  if (title === undefined) {
+    title = siteName;
+  } else {
+    title = `${title} | ${siteName}`;
+  }
```

Comment:
これは title に破壊的な変更を加えているので良くない書き方です。
たぶんRubyの感覚で書いたのだと思いますが、近年のフロントエンド開発や静的型付け言語では、イミュータビリティを重視する文化があります。
参考記事: [イミュータブルが大事な理由](https://zenn.dev/luvmini511/articles/85e8e3c71a2f41)

これを改善するには、2つほどやり方があって、

1. 別の変数に代入する

```suggestion
  const pageTitle = title === undefined ? siteName : `${title} | ${siteName}`;
```

2. タイトルを変換する関数を別の場所に（コンポーネントの前後に）定義する

```typescript
const generatePageTitle = (title) => {
  const siteName = "Kintai";
  if (title === undefined) {
    return siteName;
  }
  return `${title} | ${siteName}`;
}
```

今回、比較的単純なので、前者のやり方で良さそうですが、複雑なロジックの場合は後者のやり方が役立つこともあるので、覚えておきましょう

Reply:
https://github.com/tatata05/next-app/pull/2/commits/3654caca5c4c08aa6dfe439d2382b2a3cae9f4be
修正しました！